src/common/virtual_file_system.cpp

```diff
@@ -14,19 +14,25 @@ unique_ptr<FileHandle> VirtualFileSystem::OpenFile(const string &path, FileOpenF
 	auto compression = flags.Compression();
 	if (compression == FileCompressionType::AUTO_DETECT) {
 		// auto detect compression settings based on file name
+		compression = FileCompressionType::UNCOMPRESSED;
 		auto lower_path = StringUtil::Lower(path);
 		if (StringUtil::EndsWith(lower_path, ".tmp")) {
 			// strip .tmp
 			lower_path = lower_path.substr(0, lower_path.length() - 4);
 		}
-		if (StringUtil::EndsWith(lower_path, ".gz")) {
```

Comment:
When adding the `.wal` suffix we explicitly [consider GET parameters by looking for a question mark](https://github.com/duckdb/duckdb/blob/bd2b98175392833aca33a613d682bb36fc4b6d0e/src/storage/storage_manager.cpp#L92). I think that's a bit cleaner than looking through the entire string for `.gz`. Perhaps we can do something similar here?